### DISCUSSION TITLE
How can I prevent double clicks on buttons to avoid concurrent API call issues?

### DISCUSSION THREAD

**Discussion by Surthika** [microsoft/microsoft-ui-xaml#9445] [Q&A]
How can we efficiently handle preventing double clicks on buttons in our app, especially when users tend to double-tap buttons for instant results? We're currently using the IsHitTestVisibleproperty to disable button clicks during actions(Ex: for hitting api calls) ,after the action completes we will enable them to perform further actions, but using this for each and every button in app is unmanageable. It adds additional property just to control button click actions.

**For Reference:**

c# code

![image](https://github.com/microsoft/microsoft-ui-xaml/assets/113254401/ffd4dbc9-cca7-472b-99e8-ff22a9ad2aba)

Xaml code:

![image](https://github.com/microsoft/microsoft-ui-xaml/assets/113254401/8aa2f8b7-b7c9-4981-ac98-a24a60e09c97)

Are there any suggestions for a more streamlined approach? Alternatively, if it's possible to utilize template controls, kindly provide guidance on how to implement them.

**Comment by kmgallahan** [comment]
Use commands, they auto-disable buttons while executing:

https://learn.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.primitives.buttonbase.command

https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/generators/relaycommand
